src/main/java/ru/ayaz/Chat.java

```diff
@@ -0,0 +1,84 @@
+package ru.ayaz;
+
+
+import ru.ayaz.ru.ayaz.exceptions.InvalidNicknameException;
+import ru.ayaz.ru.ayaz.exceptions.RegistrationFailException;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.HashMap;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+
+public class Chat {
+
+    private ServerSocket serverSocket;
+    private UserMessageDistributor messageDistributor;
+
+    Chat() {
+        try {
+            this.serverSocket = new ServerSocket(4400);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+
+    void start() {
+        startUserMessageDistributor();
+        try {
+            while (true) {
+                Socket socket = serverSocket.accept();
+                UserSocketHandler userSocketHandler = new UserSocketHandler(socket, messageDistributor);
+                User user = registerUser(userSocketHandler.getWriter(), userSocketHandler.getReader());
```

Comment:
1) If we mix User and UserSocketHandler, then we will lose "controller-logic" structure. I thought that User, AdminUser, Room and etc classes can be refered to AppLogic.

2) I think registerUser method should stay in Chat class.